aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageUploadInputStreamOperation.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3.operation;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.hub.HubChannel;
+import com.amplifyframework.hub.HubEvent;
+import com.amplifyframework.storage.StorageChannelEventName;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.operation.StorageUploadFileOperation;
+import com.amplifyframework.storage.result.StorageTransferProgress;
+import com.amplifyframework.storage.result.StorageUploadFileResult;
+import com.amplifyframework.storage.s3.CognitoAuthProvider;
+import com.amplifyframework.storage.s3.ServerSideEncryption;
+import com.amplifyframework.storage.s3.request.AWSS3StorageUploadInputStreamRequest;
+import com.amplifyframework.storage.s3.service.StorageService;
+import com.amplifyframework.storage.s3.utils.S3Keys;
+
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferListener;
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferObserver;
+import com.amazonaws.mobileconnectors.s3.transferutility.TransferState;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+
+import java.io.InputStream;
+import java.util.Objects;
+
+/**
+ * An operation to upload an inputStream from AWS S3.
+ */
+public final class AWSS3StorageUploadInputStreamOperation
```

Comment:
There's a lot of duplication in this class with the `AWSS3StorageUploadFileOperation`.  Any ideas on how to reduce it?

Reply:
I tried to make them extends the same superclass. But it makes it harder to call `getRequest()` method because they are using different types of `Request`. Now this is the only thing left to be addressed in this PR.